syft/generic/string.py

```diff
@@ -0,0 +1,301 @@
+from typing import List, Tuple
+import syft as sy
+from syft.workers.base import BaseWorker
+from syft.generic.object import AbstractObject
+import abc
+
+
+class BuiltinTypeWrapper(abc.ABCMeta):
+    """
+       This is a meta class.
+
+       It is used to create a wrapper class around a python
+       built-in type such as 'str', 'list', 'dict'.
+    """
+
+    def __new__(cls, clsname, bases, dct, wrapped_type: type):
+
+        # Set of attributes to ignore by the wrapper.
+        # Those attributes are not to be overridden.
+        ignored_attr = set(
+            [
+                "__class__",
+                "__mro__",
+                "__new__",
+                "__init__",
+                "__init_subclass__",
+                "__subclasshook__",
+                "__setattr__",
+                "__getattr__",
+                "__getattribute__",
+            ]
+        )
+
+        for attr in dir(wrapped_type):
+
+            # Get only magic attributes that are not
+            # in the 'ignored_attr' set
+            if attr not in ignored_attr:
+
+                dct[attr] = cls.method_hook(cls, attr)
+
+        return super(BuiltinTypeWrapper, cls).__new__(cls, clsname, bases, dct)
+
+    def _args_adaptor(self: object, args: Tuple[object]):
```

Comment:
You might also want to use what's done in hook_args as well which does unwrap the args (_args_adaptor) and wrap the response back (_wrap_return_value), although it might be complicated to adapt. Normally for tensors it's a one-line change in hook_args.py

Reply:
Done